### PR TITLE
CLOUDP-121254: Add registration flow to setup

### PR DESCRIPTION
**Pull request by blva**
## Proposed changes

_Jira ticket:_ [CLOUDP-121254](https://jira.mongodb.org/browse/CLOUDP-121254)

* Introduces registration flow into the setup command
* Creates an interface for RegistrationOpts so that the Run and PreRun methods can be mocked
* Adds alias for `atlas register`
## Checklist

- [ ] I have signed the [MongoDB CLA](https://www.mongodb.com/legal/contributor-agreement)
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added any necessary documentation in document requirements section listed in [CONTRIBUTING.md](https://github.com/mongodb/mongocli/blob/master/CONTRIBUTING.md) (if appropriate)
- [ ] I have addressed the @mongodb/docs-cloud-team comments (if appropriate)
- [ ] I have updated [e2e/E2E-TESTS.md](https://github.com/mongodb/mongocli/blob/master/e2e/E2E-TESTS.md) (if an e2e test has been added)
- [ ] I have run `make fmt` and formatted my code

### CODE REVIEW

File: internal/cli/auth/register.go [hunk 1]
```diff
@@ -113,14 +122,33 @@ func (opts *RegisterOpts) Run(ctx context.Context) error {
...
+}
+
+func (opts *RegisterOpts) registerPreRun() error {
+	if hasUserProgrammaticKeys() {
+		return fmt.Errorf(`you have already set the programmatic keys for this profile. 
+
+Run '%s auth register --profile <profileName>' to use your username and password with a new profile`, config.BinName())
+	}
+	return nil
+}
```
Comment (tibulca) [inline internal/cli/auth/register.go:150]: should we have a const alreadyRegisteredError = fmt.Errorf(`you have already ....) then in setup_cmd, check if the opts.register.PreRun(cmd) returns this err (which from `setup` point of view is not an error, the flow should continue)?

File: internal/cli/auth/login.go [hunk 2]
```diff
@@ -93,6 +93,7 @@ func (opts *LoginOpts) SetOAuthUpAccess() {
 }
 
 func (opts *LoginOpts) Run(ctx context.Context) error {
+	fmt.Println("here somehow")
```
[the remark this answers is not in the document]
Comment (blva) [reply internal/cli/auth/login.go:96]: ooops, thanks

File: internal/cli/auth/register.go (hunk 1 above)
Comment (blva) [reply internal/cli/auth/register.go:150]: we'll not display this error for setup, that is why I've separated this from the PreRun. This is not being used on setup.

I'll send you the link for the wording in private.

**Review by tibulca — APPROVED** [review]

File: internal/cli/auth/register.go [hunk 3]
```diff
@@ -113,16 +126,35 @@ func (opts *RegisterOpts) Run(ctx context.Context) error {
...
-	_, _ = fmt.Fprintf(opts.login.OutWriter, "Successfully logged in as %s.\n", s)
+	_, _ = fmt.Fprintf(opts.OutWriter, "Successfully logged in as %s.\n", s)
 	if opts.login.SkipConfig {
 		return opts.login.config.Save()
 	}
 
 	return nil
 }
 
+func (opts *registerOpts) PreRun(cmd *cobra.Command) error {
```
[the remark this answers is not in the document]
Comment (blva) [reply internal/cli/auth/register.go]: Great point! Updated.

**Pull request merged**